### PR TITLE
Team Creation

### LINKED ISSUE #86: BE: Team Management Routes
We'll need a few routes to:

- Create a team
- Update team info
- Add a member to a team
- Remove a member from a team
- Promote a member to an admin (or demote them from admin)

### LINKED ISSUE #90: IN: Create a Team

### PR DESCRIPTION
**Pull request by WhoaNellie**
Basic team routes laid out and team creation integrated. 

Added suggested emails based on all users in db
image

And modified UI to only send invite to members that aren't signed up w/ calendApp (email sending not yet integrated)
image
image

### CODE REVIEW

File: server/routes/teamAPI.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    .then((response) => res.send(response))
+    .catch((error) => {
+      console.log(error);
+      res.status(500).send(error);
+    });
+});
+
+// ADD Team member
+router.get("/api/team/add", auth, (req, res) => {
+  db.Team.updateOne({ _id: req.body.teamId }, { $push: req.body.member })
```
Comment (dproc96) [inline server/routes/teamAPI.js]: Shouldn't this be { $push: { members: req.body.member } })?

File: server/routes/teamAPI.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        .catch((error) => {
+          console.log(error);
+          res.status(500).send(error);
+        });
+    })
+    .catch((error) => {
+      console.log(error);
+      res.status(500).send(error);
+    });
+});
```
Comment (alvyjudy) [inline server/routes/teamAPI.js]: If time permits, you can make the chain more readable by returning a promise:
```
db.User.updateOne()
.then(()=> db.User.updateOne())
.then(()=>{res.send()})
.catch(()=>{res.send()})
```

**Review by alvyjudy — COMMENTED** [review]
Nice set up! Is it possible to include some comments on what the data and body would include?
something like `//req.body: {_id, eventId,}` `//data: {newAdminId, members}` etc.

That could save everyone time to add a console log to the route to examine what the data contains.

File: server/routes/userAPI.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
-const express = require('express');
-const db = require('../db/models');
-const auth = require('../middleware/auth');
-const mongoose = require('mongoose');
```
Comment (mattchx) [inline server/routes/userAPI.js:4]: single quotation marks causing havoc

File: server/routes/teamAPI.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      .catch((error) => {
+        console.log(error);
+        res.status(500).send(error);
+      });
+  });
+});
+
+// Change Admin
+router.post("/api/team/admin", auth, (req, res) => {
+  db.User.updateOne({ _id: req.body.newAdminId }, { $set: { isAdmin: true } })
```
Comment (mattchx) [inline server/routes/teamAPI.js:62]: I've read in the documentation that `$set` is implied here however for readability it probably is best to keep it.

File: server/routes/teamAPI.js (hunk 2 above)
Comment (mattchx) [reply server/routes/teamAPI.js]: @alvyjudy You're recommendation is to add a second `.then` onto the first `.updateOne`. And the second `.then` handles the response of the second `.updateOne`, correct? 
Would that include only one `res.status(500).send(error);`?

File: server/routes/teamAPI.js (hunk 1 above)
Comment (mattchx) [reply server/routes/teamAPI.js]: I'm thinking the same unless team members are somehow pushed directly onto the array.

File: server/routes/teamAPI.js (hunk 2 above)
Comment (alvyjudy) [reply server/routes/teamAPI.js]: Yep. Unless we choose to do something differently depending on which error we are dealing with, we can just chain the `then` and have only one `catch((error)=>{res.send(error)}` at the end. It's just to make things more readable though so not a pressing issue lol

File: server/routes/teamAPI.js (hunk 1 above)
Comment (WhoaNellie) [reply server/routes/teamAPI.js]: Yes, it should!

File: server/routes/teamAPI.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+const auth = require("../middleware/auth");
+
+const router = express.Router();
+
+// CREATE team
+// req.body: { name: name, members: [email:string] }
+router.post("/api/team/create", auth, async (req, res) => {
+  let team;
+  const findMembers = req.body.members.map((email) => {
+    return db.User.findOne({ email: email });
```
Comment (kozaktar) [inline server/routes/teamAPI.js:12]: we can use db.User.find({eamil:{$in: req.body.members}}) instead of map

**Review by kozaktar — APPROVED** [review]
Good work Nellie!

File: server/routes/teamAPI.js (hunk 5 above)
Comment (WhoaNellie) [reply server/routes/teamAPI.js:12]: Wish I knew about this before I made it an array of promises 🙃

**Pull request merged**